From cd60ea1bd88ac47856ac66266a0771478ac73bad Mon Sep 17 00:00:00 2001
From: Mahesh Bodapati <mbodapat@xilinx.com>
Date: Tue, 13 Sep 2022 11:58:14 +0530
Subject: [PATCH 22/53] [Fix, microblaze]: Fix internal compiler error with
 msmall-divides     This patch will fix the internal error
 microblaze_expand_divide function which come     of rtx PLUS where the
 mem_rtx is of type SI and the operand is of type QImode.     This patch
 modifies the mem_rtx as QImode and Plus as QImode to fix the error.

    Signed-off-by :Nagaraju Mekala <nmekala@xilix.com>
                   Ajit Agarwal  <ajitkum@xilinx.com>
Upstream-Status: Pending

Signed-off-by: Mark Hatle <mark.hatle@amd.com>

---
 gcc/config/microblaze/microblaze.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/config/microblaze/microblaze.cc b/gcc/config/microblaze/microblaze.cc
index 7c648cda1b2..907c0afa9b8 100644
--- a/gcc/config/microblaze/microblaze.cc
+++ b/gcc/config/microblaze/microblaze.cc
@@ -3768,7 +3768,7 @@ microblaze_expand_divide (rtx operands[])
   emit_insn (gen_ashlsi3_bshift (regt1, operands[1], GEN_INT(4)));
   emit_insn (gen_addsi3 (regt1, regt1, operands[2]));
   mem_rtx = gen_rtx_MEM (QImode,
-                            gen_rtx_PLUS (Pmode, regt1, div_table_rtx));
+                            gen_rtx_PLUS (QImode, regt1, div_table_rtx));
 
   insn = emit_insn (gen_zero_extendqisi2(operands[0],mem_rtx));
   jump = emit_jump_insn_after (gen_jump (div_end_label), insn); 
-- 
2.37.1 (Apple Git-137.1)

